chord_metadata_service/mcode/parse_fhir_mcode.py

```diff
@@ -0,0 +1,306 @@
+                            cancer_conditions.append(cancer_condition)
+
+        # get TNM staging stage category
+        if resource["resourceType"] == "Observation" and "meta" in resource:
+            resource_profiles = resource["meta"]["profile"]
+            stage_groups = [MCODE_TNM_CLINICAL_STAGE_GROUP, MCODE_TNM_PATHOLOGIC_STAGE_GROUP]
+            for sg in stage_groups:
+                if sg in resource_profiles:
+                    tnm_staging = {"id": resource["id"]}
+                    tnm_stage_group = observation_to_tnm_staging(resource)
+                    tnm_staging["cancer_condition"] = tnm_stage_group["cancer_condition"]
+                    tnm_staging["stage_group"] = tnm_stage_group["tnm_staging_value"]
+                    for key, value in MCODE_PROFILES_MAPPING["tnm_staging"]["properties_profile"]["stage_group"].items():
+                        if sg == value:
+                            tnm_staging["tnm_type"] = key
+                    if "hasMember" in resource:
+                        members = []
+                        for member in resource["hasMember"]:
+                            member_observation_id = member["reference"].split('/')[-1]
+                            members.append(member_observation_id)
+                        # collect all members of this staging in a dict
+                        staging_to_members[resource["id"]] = members
+                    tnm_stagings.append(tnm_staging)
+
+        # get all TNM staging members
+        if resource["resourceType"] == "Observation" and "meta" in resource:
+            primary_tumor_category = _get_tnm_staging_property(resource,
+                                                               [MCODE_TNM_CLINICAL_PRIMARY_TUMOR_CATEGORY,
+                                                                MCODE_TNM_PATHOLOGIC_PRIMARY_TUMOR_CATEGORY],
+                                                               'primary_tumor_category')
+            regional_nodes_category = _get_tnm_staging_property(resource,
+                                                                [MCODE_TNM_CLINICAL_REGIONAL_NODES_CATEGORY,
+                                                                 MCODE_TNM_PATHOLOGIC_REGIONAL_NODES_CATEGORY],
+                                                                'regional_nodes_category')
+            distant_metastases_category = _get_tnm_staging_property(resource,
+                                                                    [MCODE_TNM_CLINICAL_REGIONAL_NODES_CATEGORY,
+                                                                     MCODE_TNM_PATHOLOGIC_REGIONAL_NODES_CATEGORY],
+                                                                    'distant_metastases_category')
+
+            for category in [primary_tumor_category, regional_nodes_category, distant_metastases_category]:
+                if category:
+                    tnm_staging_members.append(category)
```

Comment:
this could be written as a list comprehension

Reply:
not sure I can do it here, the `tnm_staging_members` list is initialized before bundle parsing